StreamChatClient_v3/Controllers/ChannelController.swift

```diff
@@ -153,16 +158,13 @@ public class ChannelControllerGeneric<ExtraData: ExtraDataTypes>: Controller, De
             callback { completion?(ClientError.FetchFailed()) }
             return
         }
-        
-        state = .active
-        
-        delegateCallback {
-            $0?.controllerWillStartFetchingRemoteData(self)
-        }
+
+        // Update observing state
+        state = .localDataFetched
         
         worker.update(channelQuery: channelQuery) { [weak self] error in
             guard let self = self else { return }
-            self.delegateCallback { $0?.controllerDidStopFetchingRemoteData(self, withError: error) }
+            self.state = error == nil ? .remoteDataFetched : .remoteDataFetchFailed(ClientError(with: error))
```

Comment:
nit: If you want to go fancy you can do:
```
self.state = error.map { .remoteDataFetchFailed(ClientError(with: error)) } ?? .remoteDataFetched
```

Reply:
Nice :)
But I will probably stay classy with good old "? :"